frontend/packages/shared/src/api/mutations.ts

```diff
@@ -114,7 +117,8 @@ export const updateAppPolicy = (org: string, app: string, payload: Policy) => pu
 export const updateAppMetadata = (org: string, app: string, payload: ApplicationMetadata) => put(appMetadataPath(org, app), payload);
 export const updateAppConfig = (org: string, app: string, payload: AppConfig) => post(serviceConfigPath(org, app), payload);
 export const uploadDataModel = (org: string, app: string, form: FormData) => post<void, FormData>(dataModelsUploadPath(org, app), form, { headers: { 'Content-Type': 'multipart/form-data' } });
-export const uploadOptionList = (org: string, app: string, payload: FormData) => post<void, FormData>(optionListPath(org, app), payload, { headers: { 'Content-Type': 'multipart/form-data' } });
+export const uploadOptionList = (org: string, app: string, payload: FormData) => post<void, FormData>(optionListUploadPath(org, app), payload, { headers: { 'Content-Type': 'multipart/form-data' } });
+export const updateOptionList = (org: string, app: string, optionsListId: string, payload: Option[]) => put<OptionsLists>(optionListUpdatePath(org, app, optionsListId), payload);
```

Comment:
There is something unlogical with doing a put with a single option list towards the endpoint and being able to receive multiple option lists in response. I see in the actual endpoint that we always only return the single option list, but it still looks a bit off... :o

Reply:
I agree, but i'm unsure how to improve it when the cache we're setting is for multiple options and we need the optionListId to update it